### DISCUSSION TITLE
problem integrating a class check in lua code

### DISCUSSION THREAD

**Discussion by scalope** [community/community#119799] [Programming Help]
### Select Topic Area

Question

### Body

Hello everyone, I discovered github, I'm probably not in the right place, but I have a problem adding a class check in my LUA code for world of warcraft.
I plan to share the weakaura to help everyone find a group easily for dungeons and raids.

I am looking to add a class check if the initial message is looking for a warrior or a druid for example, we must play a class specified in the message for the code to trigger

function(states, event, incomingMessage, sender, _, _, _, _, _, _, channelName)
    local function PlayIncomingSound()
        local incomingSoundFilePath = "Interface\\AddOns\\FojjiCore\\sound\\Incoming.ogg"
        PlaySoundFile(incomingSoundFilePath, "Master")
    end    
    
    -- Fonction pour gérer les mots exclus
    -- Vérifie si le message doit être exclu
    local function containsExcludedWords(message)
        -- Vérifie si le message contient au moins un mot de la liste d'exclusion
        if exclusionPattern ~= "" then
            for _, exclusionword in ipairs(aura_env.config.exclusionword) do
                
                local wordPattern = "%f[%S]" .. exclusionword.exclusion .. "%f[%A]"
                
                if message:lower():find(wordPattern) then
                    return true -- Exclure le message si un mot de la liste d'exclusion est trouvé
                end
            end
        end
        return false
    end
    
    
    -- Définition de l'abréviation des classes
    local classAbbreviation = {
        
        ["WARLOCK"] = "warlock",
        ["WARRIOR"] = "warrior",
        ["HUNTER"] = "hunter",
        ["MAGE"] = "mage",
        ["DRUID"] = "druid",
        ["ROGUE"] = "rogue",
        ["PRIEST"] = "priest",  -- Map "Priest" vers "priest"
        ["PALADIN"] = "paladin",
        ["SHAMAN"] = "shaman",
        ["DEATHKNIGHT"] = "deathknight"
    }
    --   print("Type de configuration:", aura_env.config.type)
    
    if aura_env.config.enable == false then return end
    if not _G["blacklist"] then
        _G["blacklist"] = {}
    end
    
    -- Utilisation de dungeon_shortcut pour stocker les instances sélectionnées
    local selectedInstances = {
        ["rfc"] = aura_env.config.dungeon_shortcut[1],
        ["wc"] = aura_env.config.dungeon_shortcut[2],
        ["dm"] = aura_env.config.dungeon_shortcut[3],
        ["sfk"] = aura_env.config.dungeon_shortcut[4],
        ["bfd"] = aura_env.config.dungeon_shortcut[5],
        ["stockade"] = aura_env.config.dungeon_shortcut[6],
        ["gnomeregan"] = aura_env.config.dungeon_shortcut[7],
        ["mara.*"] = aura_env.config.dungeon_shortcut[8],
        ["zf"] = aura_env.config.dungeon_shortcut[9],
        ["brd"] = aura_env.config.dungeon_shortcut[10],
        ["temple"] = aura_env.config.dungeon_shortcut[11],
        ["efefefefefee"] = aura_env.config.dungeon_shortcut[12],
        ["incurs.*"] = aura_env.config.dungeon_shortcut[13]
    }
    
    -- Ajoutez d'autres instances ici en fonction des options
    
    if not _G["role_list"] then
        _G["role_list"] = {"tank", "heal", "dps"}
    end
    
    if incomingMessage then
        local message = string.lower(incomingMessage)
        
        -- Vérifiez si le message contient un mot exclu
        if containsExcludedWords(message) then
            return
        end
        
        if incomingMessage and (
            (aura_env.config.say and event == "CHAT_MSG_SAY")
            or (aura_env.config.yell and event == "CHAT_MSG_YELL")
            or (aura_env.config.whisper and event == "CHAT_MSG_WHISPER")
            or (aura_env.config.guild and event == "CHAT_MSG_GUILD")
            or (aura_env.config.world and string.lower(channelName) == "world")
            or ((channelName ~= nil) and (
                    ( aura_env.config.general and channelName:match("General")) )
                or (aura_env.config.lookingforgroup and channelName == "LookingForGroup")
                or (aura_env.config.trade and channelName == "Trade - City")
            )
            or (aura_env.config.all)
        ) then
            
            -- Le reste de votre logique pour traiter les messages entrants
            local role
            local allRole = ""
            local sendMessage = false
            local triggeringSender = sender -- Utiliser le nom du joueur qui a envoyé le message initial
            
            if aura_env.config.debug then 
                triggeringSender = Ambiguate(sender, "none") -- Utiliser le nom Ambiguate si le mode de débogage est activé
            end
            
            local lf_type
            if aura_env.config.lf_type == 1 then
                lf_type = "lf[%d]?m"
            else
                lf_type = "lfg"
            end
            
            if string.match(message, lf_type) or (lf_type ~= "lfg" and string.match(message, "lf%d ")) then
                for instance, selected in pairs(selectedInstances) do
                    if selected then
                        if string.match(message, instance) then
                            for index, value in pairs(aura_env.config.role) do
                                if value == true then
                                    role = _G["role_list"][index]
                                    if string.match(message, role) or (role == "dps" and string.match(message,"dd")) or (not string.match(message,"tank") and not string.match(message,"heal") and not string.match(message,"dps") and not string.match(message,"dd")) then
                                        
                                        
                                        if (aura_env.config.type == 1 and ((string.match(message, "melee") and not string.match(message, "no melee")) or string.match(message, "no caster"))) or
                                        (aura_env.config.type == 2 and ((string.match(message, "caster") and not string.match(message, "no caster")) or string.match(message, "no melee"))) or
                                        (not string.match(message, "melee") and not string.match(message, "no melee") and not string.match(message, "caster") and not string.match(message, "no caster")) then
                                            
                                            
                                            sendMessage = true
                                            allRole = allRole  .. " " .. role 
                                        end
                                    end
                                end
                            end
                            if sendMessage then
                                if aura_env.config.invite then
                                    InviteUnit(triggeringSender)
                                elseif not IsInGroup() then
                                    -- Vérifier si l'utilisateur est déjà blacklisté
                                    if not _G["blacklist"][triggeringSender] or _G["blacklist"][triggeringSender] <= GetTime()-1 then
                                        -- Pour ceux qui ont le jeu en français, renvoie la classe en anglais.
                                        local function getClassAbbreviation()
                                            local _, class = UnitClass("player")
                                            local classAbbreviation = {
                                                ["WARLOCK"] = "warlock",
                                                ["WARRIOR"] = "warrior",
                                                ["HUNTER"] = "hunter",
                                                ["MAGE"] = "mage",
                                                ["DRUID"] = "druid",
                                                ["ROGUE"] = "rogue",
                                                ["PRIEST"] = "priest",  -- Map "Priest" vers "priest"
                                                ["PALADIN"] = "paladin",
                                                ["SHAMAN"] = "shaman",
                                                ["DEATHKNIGHT"] = "deathknight"
                                            }
                                            return classAbbreviation[class] or class
                                        end
                                        
                                        local classAbbrev = getClassAbbreviation()
                                        
                                        local response_message = allRole .. " " .. classAbbrev 
                                        if aura_env.config.debug then
                                            print("" .. triggeringSender .. ": " .. incomingMessage)
                                        end
                                        SendChatMessage(response_message, "WHISPER", GetDefaultLanguage("player"), triggeringSender)
                                        PlayIncomingSound() -- Appel de la fonction pour jouer le son
                                        _G["blacklist"][triggeringSender] = GetTime()
                                    end
                                end
                            end
                        end
                    end
                end
            end
        end
    end
end

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @annonymemouse !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Code Search and Navigation category is solely related to conversations around the GitHub product Discussions CS&N. This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!

**Reply by scalope** [reply]
Okay thank you :-) and sorry for miss category
